src/views/containerNode.ts

```diff
@@ -15,23 +15,36 @@ export class ContainerNode extends DataNode {
         super(nodeData, parent);
     }
 
+    private _containerType: ContainerType;
+
     public get projectBasePath() {
         return this._project.uri && Uri.parse(this._project.uri).fsPath;
     }
 
-    public getContainerType(): string {
+    public getContainerType(): ContainerType {
+        if (this._containerType) {
+            return this._containerType;
+        }
+
         const containerPath: string = this._nodeData.path || "";
         if (containerPath.startsWith(ContainerPath.JRE)) {
-            return ContainerType.JRE;
+            this._containerType = ContainerType.JRE;
         } else if (containerPath.startsWith(ContainerPath.Maven)) {
-            return ContainerType.Maven;
+            this._containerType = ContainerType.Maven;
         } else if (containerPath.startsWith(ContainerPath.Gradle)) {
-            return ContainerType.Gradle;
+            this._containerType = ContainerType.Gradle;
         } else if (containerPath.startsWith(ContainerPath.ReferencedLibrary) && this._project.isUnmanagedFolder()) {
             // currently, we only support editing referenced libraries in unmanaged folders
-            return ContainerType.ReferencedLibrary;
+            this._containerType = ContainerType.ReferencedLibrary;
+        } else {
+            this._containerType = ContainerType.Unknown;
         }
-        return ContainerType.Unknown;
+
+        return this._containerType;
+    }
+
+    public isMavenType(): boolean {
+        return this._containerType == ContainerType.Maven;
```

Comment:
ERROR: (triple-equals) D:/a/vscode-java-dependency/vscode-java-dependency/src/views/containerNode.ts[47, 36]: == should be ===